### ISSUE TITLE
[Feature] Support health checks in `consul_service` resource

### ISSUE THREAD

**Issue by lawliet89** [terraform-providers/terraform-provider-consul#54]
### Affected Resource(s)
- `consul_service`

I would like to request for the ability to define health checks for services registered with the `consul_service` resource. 

As far as I understand, the resource is meant for use with [external services](https://www.consul.io/docs/guides/external.html) and it would be great if we can also define health checks for the external service. Then, we can use something else like [Consul ESM](https://github.com/hashicorp/consul-esm) to do the actual health checking.

**Comment by pearkes** [comment]
I agree, this would be a good addition.

**Comment by remilapeyre** [comment]
Hi @pearkes, what is the status of this proposal? I don't know go but have the same need than @lawliet89 and could try to implement it if nobody have time to spend on it.

**Comment by remilapeyre** [comment]
Hi, I tried to make a PR to add this feature, I never wrote Go before so it's probably hideous. I had to make a change in `vendor/github.com/hashicorp/consul/api/catalog.go` which is probably bad. Do you want me to make a new PR in `hashicorp/consul` ?

**Comment by pearkes** [comment]
@remilapeyre thanks for opening that PR!

Yes, for us to change the API interface that is vendored we need to update the upstream at hashicorp/consul (which you could make a separate PR for) and then re-vendor here with `govendor fetch github.com/hashicorp/consul@SHA`.  I can keep an eye out for that PR an d once that is done take a closer look here.

**Comment by pearkes** [comment]
@remilapeyre awesome I think we're closer! Once you get that updated I'll take a closer review.

**Comment by pearkes** [comment]
This is done via https://github.com/terraform-providers/terraform-provider-consul/pull/64.
